### PR TITLE
**Feature:** UI-43 Id input style and a custom status icon

### PR DESCRIPTION
**Pull request by mpotomin**
# ToDo:
- [x] Add tests
- [x] Swap the id Icon to use the new one 

# Summary
1. Adds Id input style with an icon
1. Ensures that the icon still works with the clear callback and the button layout
1. Adds the possibility to display user-defined status icon 
1. Ensures the status icon still with fullWidth option

Screenshot 2019-06-19 at 13 45 48

# Related issue
https://contiamo.atlassian.net/browse/UI-43

# To be tested

Me
- [ ] No error or warning in the console on `localhost:6060`

Tester 1

- [ ] Things look good on the demo.
- [ ] No regressions can be found

Tester 2
- [ ] Things look good on the demo.
- [ ] No regressions can be found

### CODE REVIEW

File: src/Input/Input.Field.tsx [hunk 1]
```diff
@@ -5,20 +5,33 @@ import styled from "../utils/styled"
 import { InputProps } from "./Input"
 import InputButton from "./Input.Button"
 import { height } from "./Input.constants"
-import { NoIcon } from "../Icon/Icon"
+import { NoIcon, ServiceAccountIcon } from "../Icon/Icon"
 
 const width = 360
+const iconBoxSize = height // Fit icons into the square with the side of the input height
```
Comment (TejasQ) [inline src/Input/Input.Field.tsx]: We have a file called `Input.constants`. Maybe this and the line above it should be exported from there?

File: src/Input/Input.tsx [hunk 2]
```diff
@@ -53,6 +53,10 @@ export interface BaseProps extends DefaultProps, DefaultInputProps, Omit<React.I
   errorComponent?: React.FC<{ errorMessage: string }>
   /** Is the input read-only? */
   readOnly?: boolean
+  /** Should this input get Id input style */
+  idStyle?: boolean
```
Comment (TejasQ) [inline src/Input/Input.tsx]: I think this naming leaks implementation details and as a result isn't as declarative. We don't really care about the style imho. I'd propose calling it `isId` or `isUniqueId` or something similar and hide away the `style` part of the name.

Comment (mpotomin) [reply src/Input/Input.tsx]: Agree, will help to make an API cleaner

**Comment by mpotomin** [comment]
Addressed the PR review comments and swapped the [ID] Icon to use the new one from Nathan.

**Pull request merged**